### DISCUSSION TITLE
editor: problem matcher for oxlint errors

### DISCUSSION THREAD

**Discussion by sschneider-ihre-pvs** [oxc-project/oxc#7778] [Feature Request]
### What version of Oxlint are you using?

0.9.9

### What happened?

It would be great if there would be a problem matcher to transfer the found issues to the problems tab

**Comment by shulaoda** [comment]
I think we have this feature. What are your specific requirements? Could you describe it?

![image](https://github.com/user-attachments/assets/9187af4b-e63f-4387-8b4a-906e486ef5e5)

**Comment by sschneider-ihre-pvs** [comment]
If I want a task like a watcher task for the complete codebase for example.

**Comment by nrayburn-tech** [comment]
Does the language server support something like this? A way to request all issues for the entire workspace?

Is the Problems tab in VS Code the right place for this? From what I can tell, only problems for open files show there.  Are there any other plugins that show problems there for files that are not opened?

**Comment by DonIsaac** [comment]
We'd have to register a problemMatcher capability in our package.json

**Comment by nrayburn-tech** [comment]
> Does the language server support something like this? A way to request all issues for the entire workspace?

Any idea on this @DonIsaac ?

I can probably come up with some workarounds like identifying all files and requesting diagnostics for each, but I’m not sure if there is a better alternative.

**Comment by nrayburn-tech** [comment]
> > Does the language server support something like this? A way to request all issues for the entire workspace?
> 
> Any idea on this @DonIsaac ?
> 
> I can probably come up with some workarounds like identifying all files and requesting diagnostics for each, but I’m not sure if there is a better alternative.

Looking at the problem matcher, this probably isn’t relevant for VS Code, but I do think it is for IntelliJ. I’ll review it some more and make an IntelliJ specific issue when I know some more about it.
